crates/bevy_asset/src/asset_server.rs

```diff
@@ -647,10 +647,11 @@ mod test {
     use crate::{loader::LoadedAsset, update_asset_storage_system};
     use bevy_app::App;
     use bevy_ecs::prelude::*;
-    use bevy_reflect::TypeUuid;
+    use bevy_reflect::{TypeName, TypeUuid};
     use bevy_utils::BoxedFuture;
 
-    #[derive(Debug, TypeUuid)]
+    // FIXME: reflect only the type name
+    #[derive(Debug, TypeUuid, TypeName)]
```

Comment:
Why are we adding `TypeName` to all these types? I feel like if doing this is optional, it should happen in a separate PR, no?

Reply:
Because `Handle<T>` implements `Reflect`, `T` must now be `TypeName` too.
I put this requirement on the `Asset` trait, so custom assets must have `TypeName` too.

Reply:
Oh I see, that makes sense. Could you put that in the PR description, please? It would be helpful for reviewers to know why you need to touch so many files.